### ISSUE TITLE
Rename `controlplane.yml` key `setup` to `setup_app_workloads` for clarity

### ISSUE THREAD

**Issue by ahangarha** [shakacode/heroku-to-control-plane#96]
From #91 
---

**Comment by ahangarha** [comment]
Currently we have such configurations:
```yaml
my-app-staging:
  <<: *common
  setup:
    - gvc
    - redis
    - memcached
    - rails
    - sidekiq
```

The `setup` entry includes more than only app_workloads!
Is this intended? If yes, what about other workloads?

**Comment by ahangarha** [comment]
Based on the description of the `setup` command, I think we shouldn't touch the `setup` entry in the config file:

Description:
> - Creates an app and all its workloads
> - Specify the templates for the app and workloads through `setup` in the `.controlplane/controlplane.yml` file


@justin808 @rafaelgomesxyz

**Comment by rafaelgomesxyz** [comment]
Wait, I'm confused, you don't think we should change the `setup` entry, but you're suggesting a different name for it?

But in terms of the name, I like `setup_templates`, because yeah, it's not just workloads, and the names have to match the names of the files in the `templates` folder.

**Comment by ahangarha** [comment]
Based on the `setup` command description and examples, the `setup` entry in the config file can contain any workload, not just app workloads. If this is correct, renaming `setup` entry into `setup_app_template` is wrong.

Otherwise, `setup_templates` would be good for making it clear what it contains.

**Comment by rafaelgomesxyz** [comment]
Yes, the `setup` entry can contain not just any workload, but any template.

`setup_app_templates` is for "setup app" from the name of the command (`setup-app`) + "templates" from the templates folder, so I don't see how that is wrong.

**Comment by ahangarha** [comment]
Sorry for my mistake. I mixed `setup_app_workloads` and `setup_app_templates`.

I agree with `setup_app_templates`.

**Comment by ahangarha** [comment]
I made #112 to apply this but I have two proposals here:

1. Drop the `setup-app` command entirely and replace it with `--all` in `apply-template`.
   I think the value of this command is to ease the initial setup of an app with all of its templates. If the user has any specific use case, then the normal usage of `apply-template t1 t2 t3` is there.
2. We assume `setup_app_template` includes all templates unless specified in the config file. This means if the user has not made an entry in the config file, still `setup-app` can be used, and it applies all the templates. In this case, the user can white-list templates in the config file under `setup_app_template`.

I think the first approach is better because it is much simpler.

**Comment by rafaelgomesxyz** [comment]
Regarding your proposals, I don't think it's a good idea to remove the `setup-app` command. We added it specifically to differentiate the behavior from `apply-template`. One of the key differences is that, in `setup-app`, if the app already exists it raises an error.

I think what we can do instead is add an `--all` option to `apply-template`, which should be a part of #80. Then we can cover all cases:

1. When we need to do the initial setup of multiple apps with the same templates (such as review apps), we can use `setup-app` with a predefined list of templates, without worrying if a new template is added
2. When we need to apply all templates, we can use `apply-template --all` or `apply-template *` (whatever we decide)
3. When we need to apply specific templates, we can use `apply-template`

**Comment by justin808** [comment]
@rafaelgomesxyz. Yes on 1. No on 2. Why would 2 be needed if we still have setup-app?

@ahangarha 

1. this is super low priority.
2. template files contain more than a workload, such as https://github.com/shakacode/react-webpack-rails-tutorial/blob/master/.controlplane/templates/postgres.yml
3. changing file formats is sort of a breaking change, so we want to minimize it


That being said, I'm OK with changing the name 

`setup` (a verb) ===> `setup_app_templates` (a more descriptive, plural noun)


We can have a deprecation message if the old `setup` is found.

**Comment by ahangarha** [comment]
> We added it specifically to differentiate the behavior from apply-template. One of the key differences is that, in setup-app, if the app already exists it raises an error.

@rafaelgomesxyz Now it makes sense to me. Done.

---
@justin808 

> this is super low priority

Since it was under the **Now** section of #91, I took it as one of the tasks to be done.

> template files contain more than a workload,...

Exactly. This was why we chose to use `setup_app_templates` rather than `setup_app_workloads`. And I think with the same logic, we should rename `app_workloads` and `additional_workloads` into `app_templates` and `additional_templates`. This is a different issue. Just wanted to mention it since it came up.
